Dockerfile

```diff
@@ -8,16 +8,69 @@ RUN apt-get update && \
         ca-certificates \
         cmake \
         g++ \
-        libboost1.58-all-dev \
-        libssl-dev \
-        libzmq3-dev \
-        libreadline-dev \
-        libsodium-dev \
         make \
         pkg-config \
         graphviz \
         doxygen \
-        git
+        git \
+        curl \
+        libtool-bin \
+        autoconf \
+        automake
+
+WORKDIR /usr/local
+
+## Boost
+ARG BOOST_VERSION=1_66_0
+ARG BOOST_VERSION_DOT=1.66.0
+RUN curl -s -L -o  boost_${BOOST_VERSION}.tar.bz2 https://sourceforge.net/projects/boost/files/boost/${BOOST_VERSION_DOT}/boost_${BOOST_VERSION}.tar.bz2/download \
```

Comment:
That's just dangerous. Can you check against a sha256 you hardcode in that file ?